wordpress-mobile/WordPress-iOS
# Issues/5350 fix backdated draft

Fixes #5350 
This PR adds a dateModified field to our `BasePost` model and uses it in tandem with date_created_gmt to determine if a draft post should be published immediately.   A draft post will "publish immediately" when the date_created_gmt and the modified date match. This mimics the behavior in core as far as I can tell by inspecting API results but I haven't found matching code in the php. 

To test: 
Run the app after switching to the branch. 
Confirm draft posts are still correctly sorted if they are set to publish immediately. (Publish immediately should be topmost in the list)

Create a new draft post on the web and sync.
Check the post card and confirm status says "Publish Immediately"
Click the Publish button on the card. 
Confirm the post Published with the current date/time. 

Create a new draft post on the web. Backdate the post and then sync the app.
Check the post card and confirm the correct publish date is shown. 
Click the publish button on the card
Confirm the post is published with date/time that was set.

Create a new draft post on the web. Backdate the post and then sync the app.
Open post settings and choose "publish immediately" via the date picker. 
Click the update/save the post via the editor
Confirm the post is published with current date/time.

Create a new scheduled post on the web and then sync the app.
Open post settings and choose "publish immediately" via the date picker. 
Click the update/save the post via the editor
Confirm the post is published with current date/time.

Create a new scheduled post on the web and then sync the app. 
Find the card for the scheduled post and click the publish button. 
Confirm the post is published with the current date/time.

Create a new draft post and save.
In the post list, confirm the draft shows "publish immediately" in its card and not a time value.
Trash the post. 
Change the status filter to view trashed posts and confirm the trashed post shows a time value (not publish immediately) in its card.
Restore the trashed draft. 
Switch back to the draft filter and find the post.
Confirm that it is still showing a time value (not publish immediately).

Needs review: @diegoreymendez
